samples/client/petstore/java/retrofit2/src/main/java/org/openapitools/client/ApiClient.java

```diff
@@ -311,9 +311,11 @@ public ApiClient addAuthorization(String authName, Interceptor authorization) {
       throw new RuntimeException("auth name \"" + authName + "\" already in api authorizations");
     }
     apiAuthorizations.put(authName, authorization);
-    if(okBuilder != null){
-      okBuilder.addInterceptor(authorization);
+    if(okBuilder == null){
+    	throw new Exception("okBuilder is null")
```

Comment:
Use RuntimeException ? (or even better : create an ApiClientException)

Reply:
The message could be more explicit. Something like "the ApiClient was created with a built okClient so it's not possible to add an authorization interceptor to it".

Reply:
What is the advantage of creating an own ApiClientException comparing to the RuntimeException here in this case?

Reply:
See https://rules.sonarsource.com/java/RSPEC-112 . But we already have a bunch of RuntimeException so...

Reply:
Ah ok I see but as you said we already have many RuntimeExceptions so I will use a RuntimeException here if that's ok and then maybe in future you/we can change them to own Exceptions.